server/store/adapters/rdbms/drivers/postgres/dialect.go

```diff
@@ -217,20 +217,24 @@ func (postgresDialect) AttributeToColumn(attr *dal.Attribute) (col *ddl.Column,
 func (d postgresDialect) ExprHandler(n *ql.ASTNode, args ...exp.Expression) (expr exp.Expression, err error) {
 	switch ref := strings.ToLower(n.Ref); ref {
 	case "concat":
-		// need to force text type on all arguments
-		aa := make([]any, len(args))
-		for a := range args {
-			aa[a] = exp.NewCastExpression(exp.NewLiteralExpression("?", args[a]), "TEXT")
-		}
-
-		return exp.NewSQLFunctionExpression("CONCAT", aa...), nil
+		return castColumnDataToText("CONCAT", args...)
 
 	case "in":
 		return drivers.OpHandlerIn(d, n, args...)
 
 	case "nin":
 		return drivers.OpHandlerNotIn(d, n, args...)
 
+	case "like", "nlike":
```

Comment:
What about cases where the type is not ID? Should it return nothing? Is that OK?
Also, is the AttributeTypeID the only one? There are some other numeric types defined,